TypeSystem/Reflection/test.cc

```diff
@@ -327,6 +327,18 @@ struct CollectFieldValues {
   }
 };
 
+CURRENT_STRUCT(StructWithSizeTWhichShouldBeUInt64T) {
+  CURRENT_FIELD(a, uint64_t);
+  CURRENT_FIELD(b, size_t);
+};
+
+static_assert(std::is_same<decltype(std::declval<StructWithSizeTWhichShouldBeUInt64T>().a), uint64_t>::value, "");
+static_assert(std::is_same<decltype(std::declval<StructWithSizeTWhichShouldBeUInt64T>().b), uint64_t>::value, "");
+
+static_assert(std::is_same<decltype(std::declval<StructWithSizeTWhichShouldBeUInt64T>().a),
+                           decltype(std::declval<StructWithSizeTWhichShouldBeUInt64T>().b)>::value,
+              "");
+
```

Comment:
Oh, I see, I've re-mapped `size_t` inside the `CURRENT_STRUCT` implementation. Not really obvious decision imo :-/ Let's chat about it offline.